### PR TITLE
Add x-max-page-items extension

### PR DESCRIPTION
**Pull request by mpokryva**
What
----
Add extension to detect breaking changes to `x-max-page-items`.

References
----------

Test&Review
------------
* Add (incompatible)
* Remove (compatible)
* Increase (compatible)
* Decrease (incompatible)

* Ensure breaking changes are not reported for major version changes, where they're allowed. This may already be handled by `prototool`.
* Lint section in `prototool.yaml` can be disregarded, as we're not using that functionality yet.
* `prototool.yaml` can probably be trimmed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -130,6 +130,12 @@
             <artifactId>lombok</artifactId>
             <version>1.18.10</version>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
```
Comment (codyaray) [inline pom.xml]: why'd you need this?

File: src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Test
+    void testDecreasedMax() {
+        assertOpenApiBackwardIncompatible(OPENAPI_DOC2, OPENAPI_DOC1);
+    }
+
+    @Test
+    void testRemovedMax() {
+        assertOpenApiBackwardIncompatible(OPENAPI_DOC2, OPENAPI_DOC3);
```
Comment (codyaray) [inline src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java]: wouldn't removing the max be compatible? its the same as increasing max to infinity

**Review by codyaray — APPROVED** [review]
lgtm once you fix the one definition... otherwise, this is good afaict

do they accept extensions like this upstream? i'm guessing not. :)

File: src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java (hunk 2 above)
Comment (mpokryva) [reply src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java]: Didn't we talk about how some clients could rely on the max to return a certain amount of elements? I don't remember what came out of that discussion

File: pom.xml (hunk 1 above)
Comment (mpokryva) [reply pom.xml]: No idea... JetBrains added it automatically and I don't know anything about Java, so I assumed it was fine (and trusted you to comment on it if necessary :)) I can remove it.

File: src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java (hunk 2 above)
Comment (codyaray) [reply src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java]: that may be true for page_size but i don't think its true for x-max-page-items

Comment (mpokryva) [reply src/test/java/com/qdesrame/openapi/test/MaxPageItemsDiffTest.java]: Ah ok

**Pull request merged**